File: scratch/replace_bcp.py
import os
import re

dirs_to_process = [
    r"C:\Users\HADE\Downloads\CAJA-TACNA---HADE-main\bancobcp",
    r"C:\Users\HADE\Downloads\CAJA-TACNA---HADE-main\bancobcp-para asesores"
]

replacements = [
    (re.compile(r'Banco BCP', re.IGNORECASE), 'Caja Tacna'),
    (re.compile(r'banco bcp', re.IGNORECASE), 'caja tacna'),
    (re.compile(r'bancobcp', re.IGNORECASE), 'cajatacna'),
    (re.compile(r'BancoBcp'), 'CajaTacna'),
    (re.compile(r'\bBCP\b'), 'CAJATACNA'),
    (re.compile(r'\bbcp\b'), 'cajatacna'),
]

exclude_dirs = {'.git', 'build', '.dart_tool', 'Pods'}

def process_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
    except UnicodeDecodeError:
        return False # Skip non-text or binary files

    original_content = content
    for pattern, repl in replacements:
        content = pattern.sub(repl, content)

    if content != original_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        return True
    return False

def main():
    modified_count = 0
    for d in dirs_to_process:
        for root, dirs, files in os.walk(d):
            # modify dirs in-place to skip excluded directories
            dirs[:] = [dir_name for dir_name in dirs if dir_name not in exclude_dirs]
            
            for file in files:
                # skip some known binaries just in case
                if file.endswith(('.png', '.jpg', '.jpeg', '.zip', '.exe', '.dll', '.so', '.dylib', '.ico')):
                    continue
                
                filepath = os.path.join(root, file)
                if process_file(filepath):
                    print(f"Modified: {filepath}")
                    modified_count += 1
                    
    print(f"Total files modified: {modified_count}")

if __name__ == "__main__":
    main()

File: scratch/replace_bcp.dart
import 'dart:io';

final List<String> dirsToProcess = [
  r"C:\Users\HADE\Downloads\CAJA-TACNA---HADE-main\bancobcp",
  r"C:\Users\HADE\Downloads\CAJA-TACNA---HADE-main\bancobcp-para asesores",
];

final List<Map<String, dynamic>> replacements = [
  {'pattern': RegExp(r'Banco BCP', caseSensitive: false), 'repl': 'Caja Tacna'},
  {'pattern': RegExp(r'banco bcp', caseSensitive: false), 'repl': 'caja tacna'},
  {'pattern': RegExp(r'bancobcp', caseSensitive: false), 'repl': 'cajatacna'},
  {'pattern': RegExp(r'BancoBcp'), 'repl': 'CajaTacna'},
  {'pattern': RegExp(r'\bBCP\b'), 'repl': 'CAJATACNA'},
  {'pattern': RegExp(r'\bbcp\b'), 'repl': 'cajatacna'},
];

final Set<String> excludeDirs = {'.git', 'build', '.dart_tool', 'Pods'};
final Set<String> excludeExtensions = {'.png', '.jpg', '.jpeg', '.zip', '.exe', '.dll', '.so', '.dylib', '.ico'};

void main() async {
  int modifiedCount = 0;

  for (final dirPath in dirsToProcess) {
    final dir = Directory(dirPath);
    if (!await dir.exists()) continue;

    final entities = dir.listSync(recursive: true, followLinks: false);

    for (final entity in entities) {
      if (entity is! File) continue;

      bool skip = false;
      for (final exclude in excludeDirs) {
        if (entity.path.contains(Platform.pathSeparator + exclude + Platform.pathSeparator) ||
            entity.path.endsWith(Platform.pathSeparator + exclude)) {
          skip = true;
          break;
        }
      }
      if (skip) continue;

      final ext = entity.path.substring(entity.path.lastIndexOf('.')).toLowerCase();
      if (excludeExtensions.contains(ext)) continue;

      try {
        final originalContent = await entity.readAsString();
        String content = originalContent;

        for (final r in replacements) {
          content = content.replaceAllMapped((r['pattern'] as RegExp), (match) => r['repl'] as String);
        }

        if (content != originalContent) {
          await entity.writeAsString(content);
          print('Modified: ${entity.path}');
          modifiedCount++;
        }
      } catch (e) {
        // Skip files that can't be read as text
      }
    }
  }

  print('Total files modified: $modifiedCount');
}
